core/shared/src/main/scala/result.scala

```diff
@@ -170,6 +170,17 @@ sealed abstract class Result[+T, E <: Exception](val answer: T, val errors: Seq[
       case Unforeseen(e) => throw e
     }
 
+  /** Filter on the answer of this result. */
+  def filter(p: T => Boolean): Result[T, E] =
+    this match {
+      case Errata(_) => this
+      case Answer(b) => if(p(b)) this else Errata[T, E](Nil)
```

Comment:
I agree. Really since `Result` is `Answer` biased for `filter` and `withFilter` to be useful in a monadic computation `filter` here in the case of `false` should return the `Monoid.zero` of `Errata`. What that is is subject to interpretation and I thought it was `Errata(Nil)` if we were to return something like `Errata(NotMatchingFilter)`,... would we provide some kind of context as to what filter failed? Since you can potentially have `Answer(1) filter (_ == 0) filter (_ == 1)`?